rpcserver.go

```diff
@@ -1163,9 +1163,26 @@ func (r *rpcServer) MarshalChainAsset(ctx context.Context, a asset.ChainAsset,
 		return nil, err
 	}
 
-	return taprpc.MarshalChainAsset(
+	rpcAsset, err := taprpc.MarshalChainAsset(
 		ctx, a, decDisplay, withWitness, keyRing,
 	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure block timestamp is set for on-chain assets.
+	if rpcAsset.ChainAnchor != nil &&
+		rpcAsset.ChainAnchor.BlockHeight != 0 &&
+		rpcAsset.ChainAnchor.BlockTimestamp == 0 {
+
+		ts := r.cfg.ChainBridge.GetBlockTimestamp(
+			ctx,
+			rpcAsset.ChainAnchor.BlockHeight,
+		)
+		rpcAsset.ChainAnchor.BlockTimestamp = ts
+	}
```

Comment:
This block doesn't feel quite right here, but I added it because I saw that the `block_timestamp` field was `0` when running `tapcli assets list`. I didn't see how to grab the timestamp from within the assets store [dbAssetsToChainAssets](https://github.com/jamaljsr/taproot-assets/blob/leaves-chain-anchor/tapdb/assets_store.go#L829-L841) function. I didn't feel like it would be appropriate to somehow call `ChainBridge.GetBlockTimestamp()` from the db level. Please let me know if there's a better way to do this.

Reply:
> I didn't see how to grab the timestamp from within the assets store dbAssetsToChainAssets function since it isn't stored in the D

Yeah the timestamp isn't stored in the DB, as we only store the transaction and not _also_ the header. 

However the header is stored in the `proof.Proof` struct, which is a state transition proof (new txn properly followed all the rules). Looks like you start to decode the proof below here in the diff, so you should be able to get the header, then the timestamp from that.

Reply:
https://github.com/lightninglabs/taproot-assets/blob/db28a03c23ed908dd7cb7ae5e411e921cdc421c0/proof/proof.go#L243-L246

Reply:
https://pkg.go.dev/github.com/btcsuite/btcd/wire#BlockHeader

Reply:
Nice! I was able to fetch the proof and grab the timestamp from the block header. Thanks for the pointer. This implementation feels much better.

Reply:
Eww! Looks like this change broke some of the tests. Going to try to resolve those issues.